cachito/web/api_v1.py

```diff
@@ -35,12 +40,41 @@ def get_request(request_id):
     :raise NotFound: if the request is not found
     :raise ValidationError: if the request ID is of invalid format
     """
-    try:
-        request_id = int(request_id)
-    except ValueError:
-        raise ValidationError(f'{request_id} is not a valid request ID')
-    request = Request.query.get_or_404(request_id)
-    return flask.jsonify(request.to_json())
+    return flask.jsonify(_get_request_obj(request_id).to_json())
+
+
+@api_v1.route('/requests/<request_id>/download', methods=['GET'])
+def download_archive(request_id):
+    """
+    Download archive of source code.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask send_file response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    :raise ValidationError: if the request ID is of invalid format
+    """
+    request = _get_request_obj(request_id)
+    # TODO: Verify request has already been processed.
+
+    app_archive_path = Git(request.repo, request.ref).archive_path
```

Comment:
This will require the API to have `cachito_archives_dir` configured in its Celery config, which I was hoping would be an implementation detail for the worker. Couldn't `fetch_gomod_source` just do this?

Reply:
You're right. I got too into the weeds and forgot about this nasty detail. 

I'll play with it in the morning. Maybe fetch app source also handles the copy cache parameter. Or maybe we introduce a task just for generating the final archive tarball. I like the second one.

Reply:
Done! This is now done on the `fetch_app_source` task itself.